cadence/contracts/services/AxelarGovernanceService.cdc

```diff
@@ -0,0 +1,296 @@
+        access(all) fun executeApp(commandResource: &AxelarGateway.CGPCommand, sourceChain: String, sourceAddress: String, payload: [UInt8]): AxelarGateway.ExecutionStatus{
+            if (sourceChain != AxelarGovernanceService.governanceChain || sourceAddress != AxelarGovernanceService.governanceAddress){
+                panic("Not Governance")
+            }
+
+
+            // let commandSelector = payload[0]
+            // let target = Address.fromBytes(payload[1])
+            // let proposedCode = String.fromUTF8(payload[2])!
+            // let timeToExecute = UInt64.fromString(String.fromUTF8(payload[3])!)!
+            // let contractName = String.encodeHex(payload[4])
+
+            //defining as constants untill abiencode/decode is included
+            let commandSelector = AxelarGovernanceService.SELECTOR_SCHEDULE_PROPOSAL
+            let target: Address= AxelarGovernanceService.gateway
+            let proposedCode = String.fromUTF8(payload)!
+            let timeToExecute = 0 as UInt64
+            let contractName = "AxelarGateway"
+
+            AxelarGovernanceService._processCommand(commandSelector: commandSelector, proposedCode: proposedCode, target: target, timeToExecute: timeToExecute, contractName: contractName)
+            return AxelarGateway.ExecutionStatus(
+                isExecuted: true,
+                statusCode: 0,
+                errorMessage: ""
+            )
+        }
+    }
+
+    //Process Command coming from Gateway
+    access(contract) fun _processCommand(commandSelector: [UInt8] ,proposedCode: String, target: Address, timeToExecute: UInt64, contractName: String){
+        let proposalHash = String.encodeHex(self.createProposalHash(proposedCode: proposedCode, target: target))
+        if (commandSelector == self.SELECTOR_SCHEDULE_PROPOSAL){
+            self.proposals[proposalHash] <-! create Proposal(id: proposalHash, proposedCode: proposedCode, target:target, contractName: contractName, timeToExecute: timeToExecute)
+            emit ProposalScheduled(
+                proposalHash: proposalHash,
+                target: target,
+                proposedCode: proposedCode,
+                eta: timeToExecute
+            )
+        }else if (commandSelector == self.SELECTOR_CANCEL_PROPOSAL){
+            destroy <- self.proposals.remove(key: proposalHash)
+            
+            emit ProposalCancelled(
+                proposalHash: proposalHash,
+                target: target,
+                proposedCode: proposedCode,
+                eta: timeToExecute
+            )
+        }else{
+            panic("InvalidCommand")
+        }
+    }
+
+    access(all) fun createNewUpdater(account: Capability<&AuthAccount>): @Updater{
+        let updater <- create Updater(authCapability: account)
+        return <-updater
+    }
+
+    access(self) fun claimAuthCapability(provider: Address): &Updater? {
```

Comment:
Thinking about this, I believe the safer way to transfer update access is to claim an `Updater` Capability instead of an `AuthAccount` Capability. Reason being, the publishing account could mistype the receiver when publishing the Capability and the blast radius of a Gateway giving an unknown party access scoped to contract updates compared to unrestricted account access is far smaller. The risk in flight is far minimized by restricting access before the transfer occur, avoiding any question about the sorts of activity that can occur post publishing by the recipient of the AuthAccount Capability.

This change would entail maintaining Updater Capabilities instead of resources in the `updaters` contract field and claiming those here instead of `AuthAccount` Capabilities. The publisher would maintain the resource while this contract maintains a Capability on said Updater.

Reply:
Is there a specific capability called Updater?  or should we create a separate capability in the gateway account called Updater that utilizes the update_expirimental.  

I agree with this approach, we originally went with AuthAccount to simplify the solution and follow the example repo as closely as possible.  But I see the security risk that comes with it.

Reply:
There isn't a special typed Updater Capability. When I say 'Updater Capability' I'm referring to a Capability on the Updater resource, so `Capability<&Updater>`

I realized the security risk as I typed the comment above, sorry about any misguidance there. I'll update that repo to reflect the pattern we're discussing here in a bit